Solutions/Corvus.Monitoring.Instrumentation.Abstractions.Specs/FakeInstrumentationSinks.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace Corvus.Monitoring.Instrumentation.Abstractions.Specs
+{
+    using System;
+    using System.Collections.Generic;
+    using Corvus.Monitoring.Instrumentation.Abstractions.Specs.Fakes;
+    using Microsoft.Extensions.DependencyInjection;
+
+    /// <summary>
+    /// Provides fake implementations of the generic and non-generic instrumentation interfaces to
+    /// tests that need them.
+    /// </summary>
+    public class FakeInstrumentationSinks
+    {
+        private readonly List<OperationDetail> operations = new List<OperationDetail>();
+        private readonly List<ExceptionDetail> exceptions = new List<ExceptionDetail>();
+        private readonly List<OperationDetail> genericOperations = new List<OperationDetail>();
+        private readonly List<ExceptionDetail> genericExceptions = new List<ExceptionDetail>();
+
+        /// <summary>
+        /// Gets a list of the operations supplied to the underyling (non-generic) fake
+        /// <see cref="IOperationsInstrumentation"/> implementation.
+        /// </summary>
+        public IReadOnlyList<OperationDetail> Operations => this.operations;
+
+        /// <summary>
+        /// Gets a list of the exceptions supplied to the underyling (non-generic) fake
+        /// <see cref="IExceptionsInstrumentation"/> implementation.
+        /// </summary>
+        public IReadOnlyList<ExceptionDetail> Exceptions => this.exceptions;
+
+        /// <summary>
+        /// Gets a list of the operations supplied to the fake generic
+        /// <see cref="IOperationsInstrumentation{T}"/> implementation.
+        /// </summary>
+        public IReadOnlyList<OperationDetail> GenericOperations => this.genericOperations;
+
+        /// <summary>
+        /// Gets a list of the operations supplied to the fake generic
+        /// <see cref="IExceptionsInstrumentation{T}"/> implementation.
+        /// </summary>
+        public IReadOnlyList<ExceptionDetail> GenericExceptions => this.genericExceptions;
+
+        public void AddNonGenericImplementationsToServices(IServiceCollection services)
```

Comment:
Do you have code analysis set up correctly? I would have expected this to have documentation.

Reply:
As a general rule we tend not to set up our test projects to require documentation. That's because some test frameworks require certain test types to be public, but we will never publish a test suite as an API for someone else to use, so there's no need for it to have docs.